src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -182,23 +182,24 @@ public static IReadOnlyCollection<T> ToCollection<T>(this IEnumerable<T> sequenc
 
         public static T? FirstOrDefault<T, TArg>(this IEnumerable<T> source, Func<T, TArg, bool> predicate, TArg arg)
         {
-            if (source == null)
+            foreach (var item in source)
             {
-                throw new ArgumentNullException(nameof(source));
+                if (predicate(item, arg))
+                    return item;
             }
 
-            if (predicate == null)
-            {
-                throw new ArgumentNullException(nameof(predicate));
-            }
```

Comment:
Why remove the argument null checks? The parameters are declared non-nullable, but we have a large amount of code that isn't nullable-enabled yet.

Reply:
Such code will null-ref here anyways. No point in making the exception better as this is not public API. If we keep the null checks here should we then add them everywhere? If not why should keep them here?